devbox.go

```diff
@@ -104,9 +105,20 @@ func (d *Devbox) Plan() *plansdk.Plan {
 	basePlan := &plansdk.Plan{
 		DevPackages:     d.cfg.Packages,
 		RuntimePackages: d.cfg.Packages,
-		SharedPlan:      d.cfg.SharedPlan,
 	}
-	return plansdk.MergePlans(basePlan, planner.GetPlan(d.srcDir))
+	plannerPlans := planner.GetPlan(d.srcDir)
+	// basePlan packages field will merge with planners devPackages and runtimePackages. Merge will be appendWithSlice.
+	plan := plansdk.MergePlans(basePlan, plannerPlans)
+
+	// fields in the basePlan's SharedPlan:
+	//   if empty, will inherit the corresponding planners fields
+	//   if set, will override corresponding planners fields
+	plan.SharedPlan = d.cfg.SharedPlan
+	if err := mergo.Merge(&plan.SharedPlan, plannerPlans.SharedPlan); err != nil {
+		panic(err) // TODO: propagate error.
+	}
+
+	return plan
```

Comment:
Should this not happen in `plansdk` instead?

as is right now, `plan` and `build` will use different implementations. (In fact, build is not merging base plan at all)

Reply:
`build()` calls `generate()`, which calls `plan()`. I think all of them calls `plan()`, even `shell()` calls `plan()` too!

Reply:
my bad! I saw `d.Plan()` and miss-understood it to be `planner.GetPlan`